### PR TITLE
Remove dependency on Karabiner-Elements

### PR DESCRIPTION
**Pull request by jasonrudolph**
**tl;dr** - Hyper Mode is now triggered by [*tapping* the <kbd>option</kbd> key](https://github.com/jasonrudolph/keyboard/blob/c5d2f336d8578e3a3236ddadd39f32c744dcaad5/README.md#hyper-mode-for-quickly-launching-apps), and you'll need to use the macOS system preferences if you want to [remap <kbd>caps lock</kbd> to <kbd>control</kbd>](https://github.com/jasonrudolph/keyboard/blob/c5d2f336d8578e3a3236ddadd39f32c744dcaad5/README.md#installation).

### Motivation

A few times each day, I experience a strange issue where my keyboard stops responding for a few seconds, and then a stream of twenty or so keystrokes occur all at once. I'm not sure what's causing it, and I don't know how to reliably reproduce it (other than just using my Mac for a few hours and waiting for it to happen 😓).

I don't know if it's a bug in [Karabiner-Elements](https://github.com/tekezo/Karabiner-Elements), a bug in this repository, or a problematic interaction between my Hammerspoon customizations and Karabiner-Elements. 🤷‍♂️ However, I disabled Karabiner-Elements for a week, and I didn't experience the issue at all. I then re-enabled Karabiner-Elements, and I started seeing the issue again.

Since _most_ of this repository's customization uses Hammerspoon, and I only use Karabiner-Elements for two small(ish) things, I'm making the bittersweet decision to remove Karabiner-Elements.

#### Tap <kbd>option</kbd> to enter Hyper Mode

I previously used Karabiner-Elements to map the <kbd>right option</kbd> key to a "<kbd>hyper</kbd>" key. Hammerspoon can't distinguish between the <kbd>right option</kbd> key and the <kbd>left option</kbd> key, and it wouldn't make sense to override _both_ keys, so we need another approach for triggering Hyper Mode.

With the changes in this pull request, you'll now _tap_ <kbd>option</kbd> (AKA <kbd>alt</kbd>) to enter Hyper Mode and *then* press any shortcut to focus the associated application. For example, if you're using the [default keybindings](https://github.com/jasonrudolph/keyboard/blob/c5d2f336d8578e3a3236ddadd39f32c744dcaad5/README.md#default-app-keybindings) to open the Finder, you would:

1. Tap the <kbd>option</kbd> key (i.e., press and then release it in quick succession) to enter Hyper Mode
2. Then, press <kbd>f</kbd> for Finder

With this approach, we avoid interfering with existing macOS shortcuts and application shortcuts that make use of the <kbd>option</kbd> key, and we still have a way to define global shortcuts for launching apps.

#### Use macOS system preferences to remap <kbd>caps lock</kbd> to <kbd>control</kbd>

I previously used Karabiner-Elements to map <kbd>caps lock</kbd> key to <kbd>control</kbd>. Hammerspoon can't do that, but it's a quick, one-time change that you can make via macOS system preferences:

Open *System Preferences*, navigate to *Keyboard > Modifier Keys*, and set the <kbd>caps lock</kbd> key to <kbd>control</kbd> [[screenshot]](https://user-images.githubusercontent.com/2988/27111039-7f620442-507b-11e7-9bcf-93d46e14af13.png)

With that customization in place, you can once again tap <kbd>caps lock</kbd> for <kbd>escape</kbd> and hold <kbd>caps lock</kbd> for <kbd>control</kbd>.

### CODE REVIEW

**Comment by viddo** [comment]
Yay, I had been thinking about looking into this too. 👍 

I believe it's a bug on Karabiner-Elements, there are at least two long issues with reports:
- https://github.com/tekezo/Karabiner-Elements/issues/545 
- https://github.com/tekezo/Karabiner-Elements/issues/466

**Comment by jasonrudolph** [comment]
@viddo: Thanks for sharing links to those issues. :bow:💟

**Comment by jnunemaker** [comment]
I think this is actually better for me as well. Currently I have a profile for my laptop keyboard and my external keyboard. Merging this means the global caps => control setting, which I use to use, but removes having multiple profiles and needing to swap them. The only thing I was worried about was tap control for escape but that works through hammerspoon so this seems good to me. 👍
